pycaret/internal/pycaret_experiment/tabular_experiment.py

```diff
@@ -1106,6 +1118,28 @@ def setup(
 
             self._internal_pipeline.steps.append(("normalize", normalize_estimator))
 
+        # transformation ========================================= >>
+
+        if transformation:
+            if transformation_method == "yeo-johnson":
+                transformation_estimator = PowerTransformer(
+                    method="yeo-johnson", standardize=True, copy=True
```

Comment:
I would suggest to set `standardize=False` and add this transfomation step before normalization. So, if the user wants to normalize after this, he can just use the `normalize` parameter (see #1453)

Reply:
Additionally, we could add box-cox as an additional option

Reply:
In fact, yeo-johnson is an extension of box-cox over the real number field (see Johnson, Y. R. A. . (2000). A new family of power transformations to improve normality or symmetry. Biometrika, 87(4), 954-959. ). Therefore, I suggest that we implement the auto-selection of box-cox and yeo-johnson. When the label is all positiveuse, use box-cox to speed up the calculation, and automatically switch to yeo-johnson when the label is not all positive.

Reply:
The transformation is applied on every column in the dataset, so we would need to check if all columns are strictly positive or make two lists of columns (one strictly positive and the other not) and add a columntransformer containing two estimators, one for each feature group. Besides, it still could fail if the user wants to apply the pipeline to new data, without realizing it contains negative values in a column.

Reply:
In the spirit of simplicity, maybe we should leave it as is. Box-cox is almost never used (as far as I know) and this way we don't need to change the API. Only change the normalization part I mentioned in the first comment. What do you think?

Reply:
Yes, I totally agree with you. I didn't think about the test dataset before.

Reply:
I have revised it according to your first comment.

Reply:
BTW, there seems to be an indentation error (`transformation` et.al. should be behind `Preprocessing`). Maybe you can help review it to see if I fixed it correctly.

Reply:
looks good. accepting the PR